spec/services/whatsapp/channel_creation_service_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
+
+describe Whatsapp::ChannelCreationService do
+  let(:account) { create(:account) }
+  let(:waba_info) { { waba_id: 'test_waba_id', business_name: 'Test Business' } }
+  let(:phone_info) do
+    {
+      phone_number_id: 'test_phone_id',
+      phone_number: '+1234567890',
+      verified: true,
+      business_name: 'Test Business'
+    }
+  end
+  let(:access_token) { 'test_access_token' }
+  let(:service) { described_class.new(account, waba_info, phone_info, access_token) }
+
+  describe '#perform' do
+    before do
+      # Clean up any existing channels to avoid phone number conflicts
+      Channel::Whatsapp.destroy_all
```

Comment:
generally, this is not needed since the database clean up should clear this out automatically for each case.